daffodil-io/src/main/scala/edu/illinois/ncsa/daffodil/io/NonByteSizeCharset.scala

```diff
@@ -42,27 +42,64 @@ import java.nio.charset.CoderResult
 import java.nio.charset.CodingErrorAction
 import edu.illinois.ncsa.daffodil.util.MaybeInt
 import edu.illinois.ncsa.daffodil.util.Maybe
+import java.nio.charset.Charset
+import edu.illinois.ncsa.daffodil.processors.charset.NBitsWidthCharset
 
 /**
  * By "non byte sized" we mean some number of bits less than 8.
  */
-trait NonByteSizeCharset {
-  def bitWidthOfACodeUnit: Int // in units of bits
-  def requiredBitOrder: BitOrder
+final class NonByteSizeCharset(val bitsCharset: NBitsWidthCharset, name: String, decodeString: String,
```

Comment:
Can this be named more clearly...

NBitsWidth_BitsCharset
NBitsWidth_JavaCharset

To make it clear which one is our interface, and which is the java interface.

Reply:
Even that is confusing. More documentation is needed. NBitsWidth_JavaCharset is an implementation of the JavaCharset API but it consumes nbit characters from byteBuffers to produce charBuffer characters.